x/stake/keeper/validator.go

```diff
@@ -246,6 +246,11 @@ func (k Keeper) UpdateValidator(ctx sdk.Context, validator types.Validator) type
 	validator.BondHeight, validator.BondIntraTxCounter = k.bondIncrement(ctx, oldFound, oldValidator, validator)
 	valPower := k.updateValidatorPower(ctx, oldFound, oldValidator, validator, pool)
 	cliffPower := k.GetCliffValidatorPower(ctx)
+	cliffValExists := (cliffPower != nil)
+	var valPowerLTcliffPower bool
+	if cliffValExists {
+		valPowerLTcliffPower = (bytes.Compare(valPower, cliffPower) == -1)
```

Comment:
I'd agree if it was done once, its done 3 times though. (Enough so that I think these should be extracted into its own function) Given that the cliff validator logic has caused so many bugs, I think anything that improves code clarity here is a good idea. (One less mental step you have to take while reading the lines)

Reply:
mhmmmm ok, sure 👍